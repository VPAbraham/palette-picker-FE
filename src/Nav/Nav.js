import React from 'react';
import './Nav.scss';
import blackPlus from '../assets/images/plus_black.svg';
import refresh from '../assets/images/refresh.svg';

export const Nav = (props) => {

  const colorChange = (e) => {
    e.preventDefault();
    props.refreshColors();
  }

  return (
    <header className="Nav">
      <div>
        <h1>PALE
          <span>
            <img src={blackPlus} alt="plus sign"/>
          </span>
          <span>
            <img src={blackPlus} alt="plus sign" />
          </span>
          E</h1>
        <h1>PICKER</h1>
      </div>
      <section className="nav-controls">
        <img className="refresh" 
        src={refresh} 
        alt="refresh icon" 
        onClick={event => colorChange(event)}
        />
        <img 
        className="add-project" 
        src={blackPlus} 
        alt="black plus symbol"
        onClick={props.openModal}
        />
      </section>
    </header>
  )
}

export default Nav;
