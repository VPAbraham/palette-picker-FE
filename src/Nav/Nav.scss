@import url('https://fonts.googleapis.com/css?family=Montserrat&display=swap');

.Nav {
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 25px;
  font-weight: 100;
  display: flex;
  justify-content: space-between;

  div {
    margin: 3em 5em;

    h1 {
      color: rgb(189, 189, 189);
      font-size: 2em;
      text-align: left;
      line-height: 55px;

      span {
        img {
          width: 1em;
        }
      }
    }
  }



  section {
    margin: 3em 5em;
    display: flex;
    width: 15%;
    justify-content: space-between;

    img {
      width: 4em;
    }
  }
}