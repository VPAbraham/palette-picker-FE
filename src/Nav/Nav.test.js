import React from 'react';
import { Nav } from './Nav';
import { shallow } from 'enzyme';

describe('Nav', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Nav
      openModal={jest.fn()}
      refreshColors={jest.fn()}
    />)
  });
  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});