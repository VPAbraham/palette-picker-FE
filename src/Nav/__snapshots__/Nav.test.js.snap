// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot with all data passed in 1`] = `
<header
  className="Nav"
>
  <div>
    <h1>
      PALE
      <span>
        <img
          alt="plus sign"
          src="plus_black.svg"
        />
      </span>
      <span>
        <img
          alt="plus sign"
          src="plus_black.svg"
        />
      </span>
      E
    </h1>
    <h1>
      PICKER
    </h1>
  </div>
  <section
    className="nav-controls"
  >
    <img
      alt="refresh icon"
      className="refresh"
      onClick={[Function]}
      src="refresh.svg"
    />
    <img
      alt="black plus symbol"
      className="add-project"
      onClick={[MockFunction]}
      src="plus_black.svg"
    />
  </section>
</header>
`;
